docs/nx-cloud/reference/custom-images.md

```diff
@@ -0,0 +1,60 @@
+# Custom images for Nx Agents
+
+{% callout type="warning" title="Nx Enterprise" %}
+Using a custom base image for Nx Agents is a feature of Nx Enterprise plan for Nx Cloud. If you're interested in Nx Enterprise, we'd love to [chat with you about enterprise](/enterprise)
+{% /callout %}
+
+Images need to have a base setup in order to run properly with Nx Agents. There are two ways to go about setting up your base image.
+It's recommended to extend our base image. If you cannot extend our image, manually configure your image with the required steps.
+
+{% tabs %}
+{% tab label="Extend image (recommended)" %}
+
+**Do not** override the `USER`, `WORKDIR` or `ENTRYPOINT` from the base image.
+
+```yaml {% fileName="nx-agent-base-image.dockerfile"}
+FROM us-east1-docker.pkg.dev/nxcloudoperations/nx-cloud-enterprise-public/nx-agents-base-images:ubuntu22.04-node20.19-v2 as base
+
+# Add any steps you need such as installing required software
+RUN sudo apt-get install....
+RUN sudo chmod ....
+```
+
+{% /tab %}
+{% tab label="Manual setup" %}
+
+> It's recommended to use Ubuntu as the base
```

Comment:
I would say it's **required** - it's not been tested with anything other than ubuntu 

we can add that if they need to use a non-ubuntu based image they can reach out again to their DPE, as we might need to help out with changing the shell etc.